plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java

```diff
@@ -178,7 +188,11 @@ void processUndeployModelResponseAndUpdate(
                         updateDocument.put(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD, newPlanningWorkerNodes.size());
                         deployToAllNodes.put(modelId, false);
                     }
-                    updateRequest.index(ML_MODEL_INDEX).id(modelId).doc(updateDocument);
+                    UpdateDataObjectRequest updateRequest = UpdateDataObjectRequest
+                        .builder()
+                        .id(modelId)
```

Comment:
shouldn't we add `tenantId` here?

Reply:
It's added globally on line 158 which overrides this.

Reply:
Removed global tenantID and added here.